haas/api.py

```diff
@@ -263,6 +263,19 @@ def node_power_off(node):
     node.obm.power_off()
 
 
+@rest_call('PUT', '/node/<node>/boot_device', Schema({
+    'node': basestring, 'bootdev': basestring,
+}))
+def node_set_bootdev(node, bootdev):
+    auth_backend = get_auth_backend()
+    node = _must_find(model.Node, node)
+    if node.project is None:
+        auth_backend.require_admin()
+    else:
+        auth_backend.require_project_access(node.project)
+    node.obm.set_bootdev(bootdev)
```

Comment:
I like having the api call do most of the lifting re: validation. What if we added an abstract method to Obm modeled after the network allocators' `is_legal_channel_for`?